### PR TITLE
Route checkbox

### PR DESCRIPTION
**Pull request by jlyon1**
#Depends on #82 
+Checkbox in admin interface
+Basic implementation of route edit handler

### CODE REVIEW

**Comment by jlyon1** [comment]
Ready

File: api/routes.go [hunk 1]
```diff
@@ -125,6 +124,34 @@ func (App *API) RoutesDeleteHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (App *API) RoutesEditHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (kochman) [inline api/routes.go]: This needs to have a giant comment saying it only allows editing of a route's `enabled` flag.

File: api/routes.go [hunk 2]
```diff
@@ -125,6 +124,34 @@ func (App *API) RoutesDeleteHandler(w http.ResponseWriter, r *http.Request) {
...
+	if(err != nil){
+		fmt.Printf("lelel: %v",err);
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
+	err = App.db.Routes.Find(bson.M{"id": route.ID}).Sort("-created").Limit(1).One(&route)
+	route.Enabled = en
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return;
```
Comment (kochman) [inline api/routes.go]: trailing semicolon

File: api/routes.go [hunk 3]
```diff
@@ -125,6 +124,34 @@ func (App *API) RoutesDeleteHandler(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	err = App.db.Routes.Update(bson.M{"id": route.ID}, route)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return;
```
Comment (kochman) [inline api/routes.go]: trailing semicolon

File: model/model.go [hunk 4]
```diff
@@ -103,7 +103,7 @@ type Route struct {
 	Description    string    `json:"description"    bson:"description"`
 	StartTime      string    `json:"startTime"      bson:"startTime"`
 	EndTime        string    `json:"endTime"        bson:"endTime"`
-	Enabled        bool      `json:"enabled,string" bson:"enabled"`
+	Enabled        bool      `json:"enabled,string"			  bson:"enabled"`
```
Comment (kochman) [inline model/model.go]: This has too much space before bson. Run the whole thing through `go fmt`?

File: static/js/app.js [hunk 5]
```diff
@@ -35,6 +35,9 @@ var App ={
   updateRoutes: function(data){
     var updatedRoute = [];
     for(var i = 0; i < data.length; i ++){
+      if(data[i].enabled == "false"){
```
Comment (kochman) [inline static/js/app.js]: Should use `===` for string equality.

File: static/js/vehicles.js [hunk 6]
```diff
@@ -6,22 +6,28 @@ refresh = true;
...
+
     <b>active</b>:<input type="checkbox" v-model="active"></input><br>
     <div class = "button" @click="send" style="width: 50px;">add</div>
 
     </div>`,
     data (){
       return{
+
         ID: "",
+
```
Comment (kochman) [inline static/js/vehicles.js]: Extra space

File: admin.html [hunk 7]
```diff
@@ -19,10 +19,12 @@
 
     <div class="navbar">
       <p class="title">RPI Shuttle Tracker</p>
+      
       <a class="logout button" href="/admin/logout/">Logout</a>
       <div class="button logout" onclick="Admin.toggleDisplay();">Toggle Route Display</div>
       <div class="button logout rtb" onclick="Admin.showMapPanel();">Create Route</div>
 
+
```
Comment (kochman) [inline admin.html]: Extra space. Necessary?

File: model/model.go (hunk 4 above)
Comment (kochman) [reply model/model.go]: Also, JSON should be a boolean.

**Pull request merged**